hls-test-utils/src/Test/Hls.hs

```diff
@@ -134,10 +144,34 @@ goldenWithHaskellDoc plugin title testDataDir path desc ext act =
     act doc
     documentContents doc
 
+goldenWithHaskellDocAndRecorder
+  :: Pretty a
+  => (Recorder (WithPriority a) -> PluginDescriptor IdeState)
+  -> TestName
+  -> FilePath
+  -> FilePath
+  -> FilePath
+  -> FilePath
+  -> (TextDocumentIdentifier -> Session ())
+  -> TestTree
+goldenWithHaskellDocAndRecorder plugin title testDataDir path desc ext act =
+  goldenGitDiff title (testDataDir </> path <.> desc <.> ext)
+  $ runSessionWithServerAndRecorder plugin testDataDir
+  $ TL.encodeUtf8 . TL.fromStrict
+  <$> do
+    doc <- openDoc (path <.> ext) "haskell"
+    void waitForBuildQueue
+    act doc
+    documentContents doc
```

Comment:
This completely subsumes `goldenWithHaskellDoc`. Do we need a new function? I would vote for refactoring `goldenWithHaskellDoc` to provide a recorder.

Reply:
Exactly that was the question, then all plugins that don't use the new recorder infrastructure will have to be updated, too. I don't see a problem with that if you give a thumbs up to that idea.

Reply:
Modified as requested